File: src/main/java/com/apmosys/employeeportal/model/Kpi.java
package com.apmosys.employeeportal.model;


import javax.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import java.time.LocalDateTime;

@Entity
@Table(name = "kpi_kra")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Kpi {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    

    
    private String name;

    private String description;
 
    private String approvedBy;
    
    private String department;

    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime createdAt; 
    
    @UpdateTimestamp
    private LocalDateTime updatedAt; 

}

File: src/main/frontend/src/app/models/kpi.ts
export interface Kpi {
  id?: number;
  name: string;
  description: string;
  createdBy: number;
  approvedBy?: string;   
  createdAt?: Date;
  updatedAt?: Date;
  department?: string;      
  quarterId?: number;
  kpis: KpiItem[];
}

export interface KpiItem {
  id?: number;
  kpiText: string;     
}
